### DISCUSSION TITLE
Long-term vision of Nx and Lerna?

### DISCUSSION THREAD

**Discussion by dlqqq** [lerna/lerna#3641] [Q&A]
What is Nrwl's long-term vision for Nx and Lerna, and how will the two coexist? I understand that Lerna will continue to be **maintained**, but can users expect Lerna to be **developed** (i.e. large feature requests)? The current process of having contributors work on both Lerna and Nx separately seems to involve a lot more effort, and I don't think that process is sustainable.

I see three viable long-term visions:

1. Lerna continues to live as mostly separate from Nx, only using Nx in its current capacity. Both Lerna and Nx will continue to accept major feature requests.
  - IMO, this is the most wasteful long-term approach; it forces users to make tradeoffs about pros/cons between 2 projects. It would be better for everyone (contributors & users) if effort were to be consolidated to a single project.
2. Lerna will eventually be deprecated once Nx achieves feature parity with Lerna.
  - This seems to make the most sense purely from a pragmatic perspective concerning code maintainability; most of the Lerna code I've seen was untyped ES5. And from what I've read, a lot of the logic lacks sufficient test coverage.
3. Lerna will eventually achieve feature parity with Nx, and live on as an extension/superset of Nx. Nx will be a base build system for power users craving minimalism, and Lerna will be a superset of Nx will all the bells and whistles attached.

I apologize if this has already been communicated in the past; but it's important for users to know today. Right now, there are tradeoffs between using Lerna and using Nx; it would save everybody's time to know in advance whether they'd be better off using Nx instead of Lerna, even though Lerna is still being actively maintained.

**Comment by ricovitch** [comment]
Agree.
Find it difficult to get a clear vision on the future of these two tools and how they will complement each other in a year or two.
Since NX now seems to provide release lifecycle commands, with versioning, publishing, changelog, etc. How is lerna still relevant and what is Nrwl's vision on this ?
